apis/rbac/v1alpha1/types.go

```diff
@@ -0,0 +1,302 @@
+	xpv1.ResourceStatus `json:",inline"`
+}
+
+// +kubebuilder:object:root=true
+
+// A RoleAssignment is a managed resource that represents an RoleAssignment.
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,azure}
+type RoleAssignment struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RoleAssignmentSpec   `json:"spec"`
+	Status RoleAssignmentStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// RoleAssignmentList contains a list of RoleAssignment items
+type RoleAssignmentList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []RoleAssignment `json:"items"`
+}
+
+// Role and ADGroup are only CRDs and has no reconcile controller
+
+// An RoleSpec defines the desired state of an Role.
+type RoleSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+
+	// RoleID - The role definition ID.
+	// +kubebuilder:validation:Required
+	RoleID string `json:"roleID"`
+}
+
+// An RoleStatus represents the observed state of an Role.
+type RoleStatus struct {
+	xpv1.ResourceStatus `json:",inline"`
+}
+
+// +kubebuilder:object:root=true
+
+// A Role is a resource that represents an Role.
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,azure}
+type Role struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RoleSpec   `json:"spec"`
+	Status RoleStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// RoleList contains a list of Role items
+type RoleList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Role `json:"items"`
+}
+
+// An ADGroupSpec defines the desired state of an ADGroup.
+type ADGroupSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+
+	// GroupID - ID of ActiveDirectory group.
+	// +kubebuilder:validation:Required
+	GroupID string `json:"groupID"`
+}
+
+// An ADGroupStatus represents the observed state of an ADGroup.
+type ADGroupStatus struct {
+	xpv1.ResourceStatus `json:",inline"`
+}
+
+// Role is only CRD and have no reconcile controller
+
+// +kubebuilder:object:root=true
+
+// A ADGroup is a resource that represents an ADGroup.
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,azure}
+type ADGroup struct {
```

Comment:
My understanding is `ADGroup` is not a managed resource, i.e., we do not have the an `ExternalConnecter` and an `ExternalClient` for `ADGroup`s. Thus, we should not treat it as a managed resource and it should not have a CRD. For the currently defined referencer and selector fields for it, you may just leave the ID field. If there is an actual Azure resource corresponding to AD groups, then in a separate PR we can introduce the `ADGroup` managed resource that actually manipulates Azure AD group objects.